Tests/DataSourceProviderTests.swift

```diff
@@ -399,11 +399,11 @@ final class DataSourceProviderTests: TestCase {
         }
 
         // GIVEN: a data source editing controller
-        let tableDataSourceEditingController = TableEditingController(
-            canEditRow: { (tableView, indexPath) -> Bool in
+        let tableDataSourceEditingController = TableEditingController<FakeViewModel>(
+            canEditRow: { (item, tableView, indexPath) -> Bool in
                 return indexPath == expectedIndexPath
         },
-            commitEditing:{ (tableView, editingStyle, indexPath) in
+            commitEditing:{ (item, tableView, editingStyle, indexPath) in
                 if editingStyle == .delete {
                     if let _ = dataSourceProvider.dataSource.remove(at: indexPath) {
```

Comment:
Thanks, @jessesquires ,
I am aware of this issue (#80) , this PR doesn't intent to fix that though. I was trying to match the `ViewConfigurator` and pass the item which in case of `canEditRow: { (item, tableView, indexPath) -> Bool` can come handy.
However (#80) is kind of annoying..

> What if we pass the dataSource as inout? Is that possible?

I had some quick play around back then and I didn't find something. However I will revisit it soon and I will update the issue 👍